icu4c/source/i18n/number_skeletons.cpp

```diff
@@ -732,6 +732,7 @@ skeleton::parseStem(const StringSegment& segment, const UCharsTrie& stemTrie, Se
 
         case STEM_CURRENCY:
             CHECK_NULL(seen, unit, status);
+            CHECK_NULL(seen, perUnit, status);
```

Comment:
Can you make the equivalent change in Java?

Reply:
Done. Required a slightly different implementation because CHECK_NULL in C++ sets the seen flag, whereas in Java checkNull only checks for references.